From f4277255dcd89068c59782d2a45dc1af702f8162 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Fri, 10 Sep 2010 22:33:42 +0100
Patch-mainline: 2.6.37
References: fate#310916
Git-commit: 8c4223bee91b771782f2ec07f2c85d81cdff3ed5
Subject: [PATCH 1385/2588] drm/i915: Only call udelay() when waiting for
 clocks to stabilise

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 8c4223bee91b771782f2ec07f2c85d81cdff3ed5)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_display.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 0999d4b..749f865 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2003,8 +2003,8 @@ static void ironlake_crtc_enable(struct drm_crtc *crtc)
 	if ((temp & DPLL_VCO_ENABLE) == 0) {
 		I915_WRITE(pch_dpll_reg, temp | DPLL_VCO_ENABLE);
 		I915_READ(pch_dpll_reg);
+		udelay(200);
 	}
-	udelay(200);
 
 	if (HAS_PCH_CPT(dev)) {
 		/* Be sure PCH DPLL SEL is set */
@@ -2150,8 +2150,6 @@ static void ironlake_crtc_disable(struct drm_crtc *crtc)
 	} else
 		DRM_DEBUG_KMS("crtc %d is disabled\n", pipe);
 
-	udelay(100);
-
 	/* Disable PF */
 	I915_WRITE(pipe ? PFB_CTL_1 : PFA_CTL_1, 0);
 	I915_WRITE(pipe ? PFB_WIN_SZ : PFA_WIN_SZ, 0);
-- 
1.7.6

